Packet++/src/IcmpV6Layer.cpp

```diff
@@ -0,0 +1,199 @@
+#define LOG_MODULE PacketLogModuleIcmpV6Layer
+
+#include "IcmpV6Layer.h"
+#include "EndianPortable.h"
+#include "IPv4Layer.h"
+#include "IPv6Layer.h"
+#include "NdpLayer.h"
+#include "PacketUtils.h"
+#include "PayloadLayer.h"
+#include <memory>
+#include <sstream>
+
+// IcmpV6Layer
+
+namespace pcpp
+{
+
+ProtocolType IcmpV6Layer::getIcmpv6Version(uint8_t *data, size_t dataLen)
+{
+	if (dataLen < sizeof(icmpv6hdr))
+		return UnknownProtocol;
+
+	icmpv6hdr *hdr = (icmpv6hdr *)data;
+
+	switch (hdr->type)
+	{
+	case ICMPv6_ECHO_REQUEST:
+		return ICMPv6EchoRequest;
+	case ICMPv6_ECHO_REPLY:
+		return ICMPv6EchoReply;
+	case ICMPv6_NEIGHBOR_SOLICITATION:
+		return NDPNeighborSolicitation;
+	case ICMPv6_NEIGHBOR_ADVERTISEMENT:
+		return NDPNeighborAdvertisement;
+	}
+
+	return UnknownProtocol;
+}
+
+ICMPv6MessageType IcmpV6Layer::getMessageType() const
+{
+	return (ICMPv6MessageType)getIcmpv6Header()->type;

```

Comment:
If we convert `ICMPv6MessageType` to `enum class` we can safely do a static cast:
```cpp
return static_cast<ICMPv6MessageType>(getIcmpv6Header()->type);
```